src/utils/publishToNpm.js

```diff
@@ -63,18 +63,57 @@ async function publishToNpm({
   if (await allowNpmPublish(version)) {
     await execWithOutput('npm', ['pack', '--dry-run'])
 
-    if (opticToken) {
-      const packageInfo = await getLocalInfo()
-      const otp = await execWithOutput('curl', [
-        '-s',
-        '-d',
-        JSON.stringify({ packageInfo: { version, name: packageInfo?.name } }),
-        '-H',
-        'Content-Type: application/json',
-        '-X',
-        'POST',
-        `${opticUrl}${opticToken}`,
-      ])
+    let otp
+
+    if (opticToken || ngrokApiKey) {
+      if (opticToken) {
+        const packageInfo = await getLocalInfo()
+        
+        // push notification via optic mobile app
+        otp = await execWithOutput('curl', [
+          '-s',
+          '-d',
+          JSON.stringify({ packageInfo: { version, name: packageInfo?.name } }),
+          '-H',
+          'Content-Type: application/json',
+          '-X',
+          'POST',
+          `${opticUrl}${opticToken}`,
+        ])
+      } else if (ngrokApiKey) {
+        // 1. run a fastify app which exposes a simple web UI with a text input to get the OTP
+        // 2. expose the app via ngrok
```

Comment:
we're having an issue here whereby we can't bundle ngrok, so we need to find a way to execute ngrok at runtime, rather than build time

Reply:
see the changes to the action.yml file for details